File: quarantined_backups/1780530934300/tmp/audit_null_programs.js
const fs = require('fs');
const path = require('path');
const rag = require('../src/engine/ragEngine');

const indexPath = path.join(__dirname, '..', 'src', 'data', 'rag_index.json');
const indexRaw = fs.readFileSync(indexPath, 'utf8');
const idx = JSON.parse(indexRaw);
const nullItems = idx.filter((item) => !item || !item.program);

const cats = {
  'Dokumen administrasi': [
    /undang[- ]?undang/i,
    /surat keputusan/i,
    /keputusan/i,
    /rektor/i,
    /kepala/i,
    /sekretaris/i,
    /izin/i,
    /notulen/i,
    /putusan/i,
    /peraturan/i,
    /instruksi/i,
    /\bsk\b/i,
    /surat /i,
    /pemberitahuan/i,
    /pengumuman/i,
    /administrasi/i,
    /laporan kegiatan/i,
    /kegiatan/i,
    /proposal/i,
    /nota dinas/i,
    /nota/i,
    /persyaratan/i
  ],
  'Dokumen akreditasi': [
    /akreditasi/i,
    /ban-pt/i,
    /banpt/i,
    /peringkat akreditasi/i,
    /surat akreditasi/i,
    /sk akreditasi/i,
    /perguruan tinggi terakreditasi/i
  ],
  'Dokumen biaya': [
    /biaya/i,
    /spp/i,
    /dana pendidikan pokok/i,
    /dpp/i,
    /ukt/i,
    /uang kuliah tunggal/i,
    /uang kuliah/i,
    /pendaftaran/i,
    /gelombang/i,
    /pembayaran/i,
    /harga/i,
    /tarif/i,
    /beasiswa/i,
    /potongan/i
  ],
  'Dokumen program studi': [
    /program studi/i,
    /prodi/i,
    /jurusan/i,
    /sistem informasi/i,
    /teknologi informasi/i,
    /bisnis digital/i,
    /sistem komputer/i,
    /manajemen informatika/i,
    /desain komunikasi visual/i,
    /multimedia/i,
    /animasi/i,
    /teknologi komputer/i,
    /rekayasa perangkat lunak/i,
    /perangkat lunak/i
  ],
  'Dokumen RPL': [
    /rekognisi pembelajaran lampau/i,
    /rpl/i,
    /rekognisi pembelajaran/i,
    /pengakuan sks/i
  ],
  'Dokumen formulir': [
    /formulir/i,
    /daftar isi/i,
    /lampiran/i,
    /template/i,
    /isi data/i,
    /kolom/i,
    /pertanyaan/i,
    /survei/i,
    /kuisioner/i
  ]
};
const ordered = [
  'Dokumen program studi',
  'Dokumen RPL',
  'Dokumen biaya',
  'Dokumen akreditasi',
  'Dokumen formulir',
  'Dokumen administrasi'
];
const classified = {};
for (const key of Object.keys(cats)) classified[key] = [];
classified['Dokumen lain'] = [];

for (const item of nullItems) {
  const chunk = String(item && item.chunk ? item.chunk : '');
  const lower = chunk.toLowerCase();
  let matched = false;
  for (const cat of ordered) {
    if (cats[cat].some((re) => re.test(lower))) {
      classified[cat].push(item);
      matched = true;
      break;
    }
  }
  if (!matched) {
    classified['Dokumen lain'].push(item);
  }
}

const lines = [];
lines.push(`total_null=${nullItems.length}`);
for (const [cat, items] of Object.entries(classified)) {
  const filenames = Array.from(new Set(items.map((it) => it.filename || it.sourceFile || 'UNKNOWN'))).sort();
  lines.push(`CATEGORY: ${cat} count=${items.length}`);
  lines.push(`  filenames sample=${filenames.slice(0, 10).join(', ')}`);
}

lines.push('\nSAMPLES 100 NULL CHUNKS:');
for (let i = 0; i < Math.min(100, nullItems.length); i += 1) {
  const item = nullItems[i];
  const category = Object.entries(classified).find(([, its]) => its.includes(item));
  const catName = category ? category[0] : 'Dokumen lain';
  const filename = item.filename || item.sourceFile || 'UNKNOWN';
  const snippet = String(item.chunk || '').trim().replace(/\r?\n/g, '\\n');
  lines.push(`--- ${i + 1} id=${item.id} filename=${filename} category=${catName}`);
  lines.push(`  snippet=${snippet.slice(0, 300)}`);
}

lines.push('\nSUSPICIOUS KEYWORD NULL CHUNKS:');
const suspiciousRe = /program studi|prodi|jurusan|sistem informasi|teknologi informasi|bisnis digital|sistem komputer/i;
let suspiciousCount = 0;
for (const item of nullItems) {
  const chunk = String(item && item.chunk ? item.chunk : '');
  if (suspiciousRe.test(chunk)) {
    suspiciousCount += 1;
    if (suspiciousCount <= 50) {
      const filename = item.filename || item.sourceFile || 'UNKNOWN';
      const norm = rag.normalizeProgramLabel(chunk);
      lines.push(`id=${item.id} filename=${filename} program=${item.program} normalize=${norm}`);
    }
  }
}
lines.push(`total suspicious count=${suspiciousCount}`);

const outPath = path.join(__dirname, 'audit_null_programs.txt');
fs.writeFileSync(outPath, lines.join('\n'), 'utf8');
console.log('Written', outPath);

File: quarantined_backups/1780530934300/tmp/audit_null_programs.py
import json
import re
from pathlib import Path
import sys
sys.path.insert(0, 'src/engine')
try:
    from ragEngine import normalizeProgramLabel
except Exception as e:
    print('ERROR importing ragEngine:', e)
    raise

index_path = Path('src/data/rag_index.json')
idx = json.loads(index_path.read_text(encoding='utf-8', errors='replace'))
null_items = [item for item in idx if not item.get('program')]

cats = {
    'Dokumen administrasi': [r'undang[- ]?undang|surat keputusan|keputusan|rektor|kepala|sekretaris|izin|notulen|putusan|peraturan|instruksi|sk |surat |pemberitahuan|pengumuman|administrasi|laporan kegiatan|kegiatan|proposal|nota dinas|nota|dokumen administrasi|persyaratan'],
    'Dokumen akreditasi': [r'akreditasi|ban-pt|banpt|akreditasi|peringkat akreditasi|surat akreditasi|sk akreditasi|perguruan tinggi terakreditasi'],
    'Dokumen biaya': [r'biaya|spp|dana pendidikan pokok|dpp|ukt|uang kuliah tunggal|uang kuliah|pendaftaran|gelombang|pembayaran|harga|tarif|biaya pendidikan|beasiswa|potongan'],
    'Dokumen program studi': [r'program studi|prodi|jurusan|sistem informasi|teknologi informasi|bisnis digital|sistem komputer|manajemen informatika|desain komunikasi visual|multimedia|animasi|teknologi komputer|rekayasa perangkat lunak|perangkat lunak'],
    'Dokumen RPL': [r'rekognisi pembelajaran lampau|rpl|rekognisi pembelajaran|pengakuan sks|rekomendasi rpl|rekayasa pembelajaran'],
    'Dokumen formulir': [r'formulir|formulir pendaftaran|daftar isi|lampiran|template|isi data|kolom|pertanyaan|survei|kuisioner|pendaftaran online|pendaftaran'],
}
ordered = [
    'Dokumen program studi',
    'Dokumen RPL',
    'Dokumen biaya',
    'Dokumen akreditasi',
    'Dokumen formulir',
    'Dokumen administrasi',
]
compiled = {cat: [re.compile(p, re.I | re.UNICODE) for p in patterns] for cat, patterns in cats.items()}
classified = {cat: [] for cat in cats}
classified['Dokumen lain'] = []
for item in null_items:
    tex = item.get('chunk', '')
    text_lo = tex.lower()
    found = False
    for cat in ordered:
        if any(p.search(text_lo) for p in compiled[cat]):
            classified[cat].append(item)
            found = True
            break
    if not found:
        classified['Dokumen lain'].append(item)

output_lines = []
output_lines.append(f'total_null={len(null_items)}')
for cat, items in classified.items():
    filenames = sorted({it.get('filename') or it.get('sourceFile') or 'UNKNOWN' for it in items})
    output_lines.append(f'CATEGORY: {cat} count={len(items)}')
    output_lines.append(f'  filenames sample={filenames[:10]}')

output_lines.append('\nSAMPLES 100 NULL CHUNKS:')
for i, item in enumerate(null_items[:100], 1):
    cat = next((c for c, its in classified.items() if item in its), 'Dokumen lain')
    snippet = item.get('chunk', '').strip().replace('\n', '\\n')
    filename = item.get('filename') or item.get('sourceFile') or 'UNKNOWN'
    output_lines.append(f'--- {i} id={item.get("id")} filename={filename} category={cat}')
    output_lines.append(f'  snippet={snippet[:300]}')

keywords = [r'program studi', r'prodi', r'jurusan', r'sistem informasi', r'teknologi informasi', r'(?<!\w)bisnis digital', r'sistem komputer']
kw_re = re.compile('|'.join(keywords), re.I)
output_lines.append('\nSUSPICIOUS KEYWORD NULL CHUNKS:')
totalsusp = 0
for item in null_items:
    if kw_re.search(item.get('chunk', '')):
        totalsusp += 1
        if totalsusp <= 50:
            prog_norm = normalizeProgramLabel(item.get('chunk', '') or '')
            filename = item.get('filename') or item.get('sourceFile') or 'UNKNOWN'
            output_lines.append(f'id={item.get("id")} filename={filename} program={item.get("program")} normalize={prog_norm}')
output_lines.append(f'total suspicious count={totalsusp}')

out_path = Path('tmp/audit_null_programs.txt')
out_path.write_text('\n'.join(output_lines), encoding='utf-8')
print(f'Wrote {out_path} ({len(output_lines)} lines)')
